File: Difficulty: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:
    // Function returns the second
    // largest elements
    int getSecondLargest(vector<int> &arr) {
        // Code Here
        int c = 0;
        for(int i=0;i<arr.size()-1;i++){
            if(arr[i] == arr[i+1]){
                c++;
            }
        }
        if(c == arr.size()-1) return -1;
        sort(arr.begin(),arr.end());
       for(int i=arr.size()-1 ; i>0 ; i--){
           if(arr[i] != arr[i-1]){
               return arr[i-1];
           }
       }
       return arr[arr.size()-2];
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        Solution ob;
        int ans = ob.getSecondLargest(arr);
        cout << ans << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String[] arr1Str = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(arr1Str).mapToInt(Integer::parseInt).toArray();
            Solution ob = new Solution();
            int ans = ob.getSecondLargest(arr);
            System.out.println(ans);

            System.out.println("~");
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public int getSecondLargest(int[] arr) {
        // Code Here
        int n = arr.length;
        int mx1 = arr[0];
        int mx2 = -1;
        
        for(int i=1;i<n;i++){
            if(arr[i] > mx1){
                mx2 = mx1;
                mx1 = arr[i];
            }else if(arr[i] > mx2 && arr[i] < mx1){
                mx2 = arr[i];
            }
        }
       return mx2;
    }
}